### ISSUE TITLE
output from inter_tree and inter_intra

### ISSUE THREAD

**Issue by paternogbc** [paternogbc/sensiPhy#123]
The idea here is to organize and make decisions easier. So we can include ideas while we have them and also try some graphs until we reach an agreement =) 

# Inter_tree
### Summary
I think Caterina has developed a good way of reporting the results for that but it would follow the same logic of the plots (just tree, just intra, and both)

### Plots
The graphs would be the original regressions with the slope and confidence intervals representing these three scenarios: 1- Just intra; 2- Just tree; 3- Joined

Plot example:

***
### Plots

***
### Plots
***
### Plots
***

# Inter_intra:
### Summary
Same as `inter_tree`
### Plots
Same as `inter_tree`
***
### Plots

***
### Plots
***
### Plots
***

**Comment by paternogbc** [comment]
# Wrapper functions
These functions will simply call one of the small functions to do the job based on the type of interaction that user wants.

## intra 
`interaction_intra_phylm`
`interaction_intra_phyglm`

## tree
`interaction_tree_phylm`
`interaction_tree_phyglm`

We could also just use two types and merge the intra and tree functions into one with 7 possible interactions: intra_influ; intra_clade; intra_samp; tree_influ; tree_clade; tree_samp; tree_intra

Then we could just use:
`interaction_phylm`
`interaction_phyglm`

# Small functions 
These functions performs a single analysis interacting two types of uncertainty 

## Intra
### phylm
`interaction_intra_influ_lm`
`interaction_intra_clade_lm`
`interaction_intra_samp_lm`
`interaction_intra_tree_lm`

### phyhlm
`interaction_intra_influ_glm`
`interaction_intra_clade_glm`
`interaction_intra_samp_glm`
`interaction_intra_tree_glm`

## tree
###  phylm
`interaction_tree_influ_lm`
`interaction_tree_clade_lm`
`interaction_tree_samp_lm`
`interaction_tree_intra_lm` (equivalent to `interaction_intra_tree`

### phyglm
`interaction_tree_influ_glm`
`interaction_tree_clade_glm`
`interaction_tree_samp_glm`
`interaction_tree_intra_glm` (equivalent to `interaction_intra_tree`

See what you think. =)

**Comment by gijsbertwerner** [comment]
Ok, I think I have the basic functionality for interaction_intra_influ_phylm done now. 

Essentially I have taken the intra_phylm and the influ_phylm function and nested the two loops inside each other. Effectively, it iterates 'times' times over resimulated datasets based on intra_xx and then for each of them runs influ_phylm. Conceptually quite simple, but I think it works.  

There is one minor thing to solve still: it currently only works without transformations because intra_xx and influ_xxx deal with these in a different way (internally vs in the main call), and I haven't managed to reconcile them. Should be easy, but I have to run now. 

You can try running it with this code, if you load the 'interactions' branch. 

testmod<-interaction_intra_influ_phylm(formula = gestaLen ~ adultMass, phy = alien$phy[[1]],data=alien$data,model="lambda",
                                    y.transf = NULL,x.transf = NULL,Vy="SD_gesta",times=3)
summary(testmod)
sensi_plot(testmod)

**Comment by gijsbertwerner** [comment]
For how to present this visually/summary-wise, we need to consider how we think about this a bit more. 

I see two options: 
(1) essentially retain the old way of presenting summary/plot, i.e. calculate the on average most influential x species across the resimulated datasets. In that case you are effectively asking what is the most influential species, averaged out over all of the resimulated datasets. Visually, this would be very easy, because the old sense_plot still works find then (it just makes the CI's a bit bigger, but doesn't change anything otherwise. For summary() we would need to make a few minor changes, because there is now not a single value per species, but `times` values, so we would need to average over those. 
(2) Somehow present the most influential species per resimulation/iteration of the dataset. This would allow the user to determine if the most influential species would vary by iterations, and in what way. For instance, we could give stats like: 'it's species A in 80% of the resimulations' or something. 

I would favour (1) over (2), for a few reasons. In favour of approach (2) you can say that this more truly an 'interaction', but I think that people will ultimately be more interested in (1), because the resimulations are not 'real' data anyway. Plus approach (1) is much easier to implement, and requires only minor tweaking to what we have. Does this make sense? What do you think?

**Comment by paternogbc** [comment]
Great @gijsbertwerner! 
The transformation problem is actually a bit tricky. We have to think/find a good code solution and standardize across the package.

Regarding the summary I would go for approach 1. I guess this is what people want to know. Given the variation in data, which are the most influential species. So simulating many datasets and checking the most influential species on average sounds a simple but effective answer to this question. And also agree about the plot. For me this works good ;)

**Comment by gijsbertwerner** [comment]
Cool, sounds good. I am going to try and work on this tomorrow, but have a pretty busy day, so it might become Tuesday. Yes, let's go for (1). 

Yes, it would be good to standardise the transformation thing across functions. I have some ideas, and will try and see how far I get :-)

**Comment by gijsbertwerner** [comment]
Ok, so I have a somewhat inelegant solution for the transformation problem. You are right, it's actually really tricky. What I do now is require the user to supply the transformation as a string (`x.transf = "log"` and then I use paste.0 to recreate the formula in the initial call, and do.call within the loops to do the actual transformation there. It's very clunky, but does work.

**Comment by gijsbertwerner** [comment]
The intra_influ_phylm function works fine now, I think. I have updated the help a bit, and changed the summary function accordingly. This now gives as an output (1) the species that are (given the cutoff) most often included as being influential for slope& intercept, ranked in decreasing order of how often they are included and (2) the mean parameter value/perc-change/p-valuye for each of these species, also ordered. So, the output now averages out over the entire iterated dataset, as we discussed above. 

In principle, sensi_plot, should still work, I think but I will take a closer look still.
